lib/adjustable_income_calculator.rb

```diff
@@ -0,0 +1,99 @@
+class AdjustableIncomeCalculator
+  # https://www.ons.gov.uk/peoplepopulationandcommunity/birthsdeathsandmarriages/lifeexpectancies/bulletins/lifeexpectancyatbirthandatage65bylocalareasinenglandandwales/2015-11-04#national-life-expectancy-at-age-65
+  LIFE_EXPECTANCY = 84
+  TAX_FREE_POT_PORTION = 0.25
+  GROWTH_INTEREST_RATE = 0.03
+
+  def initialize(pot:, age:, desired_income:)
+    self.pot = pot
+    self.age = age
+    self.desired_income = desired_income
+  end
+
+  attr_accessor :pot, :age, :desired_income
+
+  def estimate
+    {
+      tax_free_lump_sum: tax_free_lump_sum,
+      taxable_portion: taxable_portion,
+      growth_interest_rate: GROWTH_INTEREST_RATE * 100,
+      monthly_drawdown_amount: (normalised_drawdown_amount / 12).to_i,
+      desired_income_with_pot_growth_lasts_until: desired_income_with_pot_growth_lasts_until.to_i,
+      monthly_income_until_life_expectancy: monthly_income_until_life_expectancy.to_i,
+      life_expectancy: LIFE_EXPECTANCY
+    }
+  end
+
+  private
+
+  def normalised_drawdown_amount
+    (desired_income > 0) ? desired_income : income_until_life_expectancy
```

Comment:
Since Ruby 2.3 numerics now have a `positive?` predicate. That would remove the operator / operand here and improve readability a teeny bit.